powerauth-webflow-authentication-form/src/test/java/io/getlime/security/powerauth/lib/webflow/authentication/method/form/encryption/AesEncryptionPasswordProtectionTest.java

```diff
@@ -0,0 +1,102 @@
+    }
+
+    @org.junit.jupiter.api.Test
+    void protect() throws NoSuchPaddingException, NoSuchAlgorithmException, InvalidAlgorithmParameterException, InvalidKeyException, BadPaddingException, IllegalBlockSizeException {
+
+        final String secretKeyBase64 = "Qee4CK44d8GduTxoHU7JPM2lCs+KF63akIpKyaLk9+c=";
+        final String cipherTransformation = "AES/CBC/PKCS7Padding";
+        final String originalPassword = "passw0rd";
+
+        System.out.println("Encrypting Password: '" + originalPassword
+                + "' with key '" + secretKeyBase64
+                + "' using '" + cipherTransformation + "' mode."
+        );
+
+        String decryptedPassword;
+
+        // Decrypt previously known password
+        String encryptedPassword = "uOaHgDWvvrHPzFSvRY6bUg==:y8SuSlUipZKkcw2QjCpuCw==";
+        decryptedPassword = decryptPassword(secretKeyBase64, cipherTransformation, encryptedPassword);
+        System.out.println("Encrypted Password - static: " + encryptedPassword);
+        System.out.println("Decrypted Password - static: " + decryptedPassword);
+        Assertions.assertEquals(decryptedPassword, originalPassword);
+
+        // Perform repeated encryption and decryption of passwords
+        AesEncryptionPasswordProtection inst = new AesEncryptionPasswordProtection(cipherTransformation, secretKeyBase64);
+        for (int i = 0; i < 10; i++){
+            encryptedPassword = inst.protect(originalPassword);
+            System.out.println("Encrypted Password " + i + ": " + encryptedPassword);
+            decryptedPassword = decryptPassword(secretKeyBase64, cipherTransformation, encryptedPassword);
+            System.out.println("Decrypted Password " + i + ": " + decryptedPassword);
+            Assertions.assertEquals(decryptedPassword, originalPassword);
+        }
+
+    }
+
+    private String decryptPassword(String secretKeyBase64, String cipherTransformation, String encryptedPassword) throws NoSuchAlgorithmException, NoSuchPaddingException, InvalidKeyException, InvalidAlgorithmParameterException, IllegalBlockSizeException, BadPaddingException {
+        // Read secret key from configuration
+        byte[] secretKeyBytes = BaseEncoding.base64().decode(secretKeyBase64);
+        SecretKey secretKey = new SecretKeySpec(secretKeyBytes, "AES");
+
+        // Extract IV and encrypted password and convert them to bytes
+        String[] parts = encryptedPassword.split(":");
+        if (parts.length != 2) {
+            throw new IllegalArgumentException("Invalid request");
+        }
+        String ivBase64 = parts[0];
+        byte[] ivBytes = BaseEncoding.base64().decode(ivBase64);
+        String encryptedPasswordBase64 = parts[1];
+        byte[] encryptedPasswordBytes = BaseEncoding.base64().decode(encryptedPasswordBase64);
+
+        // Decrypt password using specified cipher transformation, extracted IV and encrypted password bytes
+        Cipher cipher = Cipher.getInstance(cipherTransformation);
```

Comment:
Perhaps this should be: `Cipher.getInstance(cipherTransformation, "BC")`, otherwise another crypto provider could be used. I assume you wanted to use BC because the provider is initialized in the test.

Reply:
Good point - fixed.